docs/dev/how-to/send-transaction-l1-l2.md

```diff
@@ -0,0 +1,334 @@
+
+// load env file
+import dotenv from "dotenv";
+dotenv.config();
+
+// Greeter contract ABI for example
+const ABI = [
+  {
+    inputs: [
+      {
+        internalType: "string",
+        name: "_greeting",
+        type: "string",
+      },
+    ],
+    stateMutability: "nonpayable",
+    type: "constructor",
+  },
+  {
+    inputs: [],
+    name: "greet",
+    outputs: [
+      {
+        internalType: "string",
+        name: "",
+        type: "string",
+      },
+    ],
+    stateMutability: "view",
+    type: "function",
+  },
+  {
+    inputs: [
+      {
+        internalType: "string",
+        name: "_greeting",
+        type: "string",
+      },
+    ],
+    name: "setGreeting",
+    outputs: [],
+    stateMutability: "nonpayable",
+    type: "function",
+  },
+];
+
+// HTTPS RPC endpoints (from local node)
+const L1_RPC_ENDPOINT = "http://localhost:8545";
+const L2_RPC_ENDPOINT = "http://localhost:3050";
+
+const WALLET_PRIV_KEY = process.env.RICH_WALLET_PRIV_KEY || "";
+
+if (!WALLET_PRIV_KEY) {
+  throw new Error("Wallet private key is not configured in env file");
+}
+
+const L2_CONTRACT_ADDRESS =  "0x..."; //
+
+async function main() {
+  console.log(`Running script for L1-L2 transaction`);
+
+  // Initialize the wallet.
+  const l1provider = new Provider(L1_RPC_ENDPOINT);
+  const l2provider = new Provider(L2_RPC_ENDPOINT);
+  const wallet = new Wallet(WALLET_PRIV_KEY, l2provider, l1provider);
+
+  // console.log(`L1 Balance is ${await wallet.getBalanceL1()}`);
+  console.log(`L2 Balance is ${await wallet.getBalance()}`);
+
+  // retrieve L1 gas price
+  const l1GasPrice = await l1provider.getGasPrice();
+  console.log(`L1 gasPrice ${ethers.utils.formatEther(l1GasPrice)} ETH`);
+
+  const contract = new Contract(L2_CONTRACT_ADDRESS, ABI, wallet);
+
+  const msg = await contract.greet();
+
+  console.log(`Message in contract is ${msg}`);
+
+  const message = `Updated at ${new Date().toUTCString()}`;
+
+  const tx = await contract.populateTransaction.setGreeting(message);
+
+  // call to RPC method zks_estimateGasL1ToL2 to estimate L2 gas limit
+  const l2GasLimt = await l2provider.estimateGasL1(tx);
+
+  console.log(`L2 gasLimit ${l2GasLimt.toString()}`);
+
+  const baseCost = await wallet.getBaseCost({
+    // L2 computation
+    gasLimit: l2GasLimt,
+    // L1 gas price
+    gasPrice: l1GasPrice,
+  });
+
+  console.log(
+    `Executing this transaction will cost ${ethers.utils.formatEther(
+      baseCost
+    )} ETH`
+  );
+
+  const iface = new ethers.utils.Interface(ABI);
+  const calldata = iface.encodeFunctionData("setGreeting", [message]);
+
+  const txReceipt = await wallet.requestExecute({
+    contractAddress: L2_CONTRACT_ADDRESS,
+    calldata,
+    l2GasLimit: l2GasLimt,
+    refundRecipient: wallet.address,
+    overrides: {
+      // send the required amount of ETH
+      value: baseCost,
+      gasPrice: l1GasPrice,
+    },
+  });
+
+  console.log("L1 tx hash is :>> ", txReceipt.hash);
+
+  txReceipt.wait();
```

Comment:
maybe makes sense to show the result on the L2 (I mean how the greeting has changed)

Reply:
okay